hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java

```diff
@@ -138,50 +151,51 @@ public String convertToLetter(int size) {
             IOUtil.closeResource(is);
         }
 
+        ExecutorService executor = Executors.newFixedThreadPool(1);
+        Future<MCResponse> mcResponse = null;
+        ManagementCenterConfig managementCenterConfig = hazelcastNode.config.getManagementCenterConfig();
+        if (managementCenterConfig.isEnabled()) {
+            mcResponse = executor.submit(new MCRequest(hazelcastNode));
+        }
+
         //Calculate native memory usage from native memory config
         NativeMemoryConfig memoryConfig = hazelcastNode.getConfig().getNativeMemoryConfig();
         final ClusterServiceImpl clusterService = hazelcastNode.getClusterService();
         long totalNativeMemorySize = clusterService.getSize(DATA_MEMBER_SELECTOR)
                 * memoryConfig.getSize().bytes();
         String nativeMemoryParameter = (isEnterprise)
                 ? Long.toString(MemoryUnit.BYTES.toGigaBytes(totalNativeMemorySize)) : "0";
-        //Calculate connected clients to the cluster.
-        Map<ClientType, Integer> clusterClientStats = hazelcastNode.clientEngine.getConnectedClientStats();
-        RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
-        OperatingSystemMXBean osMxBean = ManagementFactory.getOperatingSystemMXBean();
 
+        RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
         Long clusterUpTime = clusterService.getClusterClock().getClusterUpTime();
+        int clusterSize = clusterService.getMembers().size();
         PhoneHomeParameterCreator parameterCreator = new PhoneHomeParameterCreator();
         parameterCreator.addParam("version", version);
         parameterCreator.addParam("m", hazelcastNode.getThisUuid());
         parameterCreator.addParam("e", Boolean.toString(isEnterprise));
-        parameterCreator.addParam("l", MD5Util.toMD5String(hazelcastNode.getConfig().getLicenseKey()));
+        String licenseKey = hazelcastNode.getConfig().getLicenseKey();
+        parameterCreator.addParam("l", licenseKey == null ? "" : MD5Util.toMD5String(licenseKey));
```

Comment:
I think that was "NULL" before and now "". Can you check with Megan if that is ok with her queries.